homeworks/01-bubble-sort/src/App.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react'
+import Button from "./components/button/Button"
+import Column from "./components/column/Column"
+import './App.css'
+
+type AppState = {
+	statusOfGame: string,
+	statusOfBtn: string,
+	arrOfNumbers: number[],
+	arrOfTimerID: number[],
+	disabled: boolean,
+}
+
+class App extends Component<{}, AppState> {
+
+  private timerID: any
+
+	state = {
+		statusOfGame: 'not solved',
+		statusOfBtn: 'Start',
+		arrOfNumbers: [],
+		arrOfTimerID: [],
+		disabled: false,
+	}
+
+	componentDidMount() {
+		this.setState({arrOfNumbers: this.makeArr()})
+	}
+
+	componentWillUnmount() {
+		const {arrOfTimerID} = this.state
+		arrOfTimerID.forEach(el => clearTimeout(el))
+	}
+
+	makeArr = (): number[] => [...Array(Math.floor(Math.random() * 40))].map(el => el = Math.floor(Math.random() * 100))
+
+	start = (): void => {
+		const {arrOfTimerID, statusOfBtn} = this.state
+		arrOfTimerID.forEach(el => clearTimeout(el))
+		this.setState({statusOfBtn: statusOfBtn === 'Start' ? 'Pause' : 'Start', disabled: false})
+		if (statusOfBtn === 'Start') {
+			this.bubbleSort()
+		}
+	}
+
+	set = (): void => {
+		const {arrOfTimerID} = this.state
+		arrOfTimerID.forEach(el => clearTimeout(el))
+		this.setState({arrOfNumbers: this.makeArr(), statusOfGame: 'not solved', statusOfBtn: 'Start', disabled: false})
+	}
+
+	bubbleSort = (): void => {
+		const arrOfNum: number[] = [...this.state.arrOfNumbers]
+		let arrOfTimerID: number[] = [];
+		let resArr: number[] = [];
+		let flagChange: boolean = false;
+		let count: number = 0
+		for(let i = 0;  i < arrOfNum.length - 1;  i++) {
+			for(let j = 1;  j < arrOfNum.length - i;  j++) {
+				if(arrOfNum[j-1] > arrOfNum[j]) {
+					count++
+					let temp = arrOfNum[j]
+					arrOfNum[j] = arrOfNum[j - 1]
+					arrOfNum[j - 1] = temp
+					resArr = [...arrOfNum]
+					flagChange = true;
+					this.timerID = setTimeout((( result, curTimerID) => {
+						return () => {
+						this.setState({arrOfNumbers : result})
+							if (this.timerID === curTimerID + 1 || result.length <= 2) {
+								this.setState({statusOfGame: 'solved', disabled: true, statusOfBtn: 'Start'})
+							}
+						}
+					})(resArr, this.timerID), 300 * count)
+					arrOfTimerID.push(this.timerID)
+				}
+			}
+		}
+		if (!flagChange) {
+			this.setState({statusOfGame: 'solved', disabled: true, statusOfBtn: 'Start'})
+		}
+		this.setState({arrOfTimerID})
+	}
```

Comment:
Would be better to keep this logic separately (eg, `utils` folder under `src`). Keep the component logic as clear as possible.

Reply:
I'll think about it, thanks.